### ISSUE TITLE
[Bug]:  DROP COLUMN unexpected error

### ISSUE THREAD

**Issue by akaError** [oceanbase/oceanbase#1035]
### Please check the issue list for the same bug

- [X] I have checked issue list and this bug is encountered for the first time.

### Environment

```markdown
observer V3.1.4
```

### Fast Reproduce Steps(Required)

CREATE TABLE v0 ( v2 INTEGER PRIMARY KEY , v1 VARCHARACTER ( 30 ) , UNIQUE v0 ( v1 ( 30 ) ) ) ;
ALTER TABLE v0 DROP COLUMN v2 ;

### Actual Behavior

<img width="893" alt="image" src="https://user-images.githubusercontent.com/97499944/183326352-bb22c5b6-d23e-4d34-b0d9-b05ee61ec2d2.png">

### Expected Behavior

it should return return not supported because this version yet do not support drop primary key column
<img width="871" alt="image" src="https://user-images.githubusercontent.com/97499944/183330041-ac4249ed-3a16-436c-b74f-2b87a3806e62.png">

**Comment by watchpoints** [comment]
what is the difference between rowkey and pimary key ?

I find pimary key is pimary key ,not rowkey ,not index 

what problem



```
int ObDDLOperator::alter_table_drop_aux_column

  int64_t rowkey_position_;        // greater than zero if this is rowkey column, 0 if this is common column

 int64_t index_position_;        // greater than zero if this is index column

// true: primary key/hidden primary key(pk_increment/cluster_id/parition_id)/partitioned key of no-pk tables

 inline bool is_rowkey_column() const

 {

  return rowkey_position_ > 0;

 }
```

**Comment by JLY2015** [comment]
yes, thank for your suggestion, we confirm that drop primary key column is not support yet, and we will return right error msg in next 3.1.5 version.

**Comment by JLY2015** [comment]
> 
In fact, both primary key and rowkey refer to the "primary key".

**Comment by JLY2015** [comment]
> what is the difference between rowkey and pimary key ?
> 
> I find pimary key is pimary key ,not rowkey ,not index
> 
> what problem
> 
> ```
> int ObDDLOperator::alter_table_drop_aux_column
> 
>   int64_t rowkey_position_;        // greater than zero if this is rowkey column, 0 if this is common column
> 
>  int64_t index_position_;        // greater than zero if this is index column
> 
> // true: primary key/hidden primary key(pk_increment/cluster_id/parition_id)/partitioned key of no-pk tables
> 
>  inline bool is_rowkey_column() const
> 
>  {
> 
>   return rowkey_position_ > 0;
> 
>  }
> ```

**Comment by JLY2015** [comment]
> what is the difference between rowkey and pimary key ?
> 
> I find pimary key is pimary key ,not rowkey ,not index
> 
> what problem
> 
> ```
> int ObDDLOperator::alter_table_drop_aux_column
> 
>   int64_t rowkey_position_;        // greater than zero if this is rowkey column, 0 if this is common column
> 
>  int64_t index_position_;        // greater than zero if this is index column
> 
> // true: primary key/hidden primary key(pk_increment/cluster_id/parition_id)/partitioned key of no-pk tables
> 
>  inline bool is_rowkey_column() const
> 
>  {
> 
>   return rowkey_position_ > 0;
> 
>  }
> ```

In fact, both primary key and rowkey refer to the "primary key".

**Comment by watchpoints** [comment]
how to change  ,if change to parse
